_posts/2024-01-30-securing-wildfly-apps-oidc-k8s.adoc

```diff
@@ -0,0 +1,398 @@
+
+. Log into the Keycloak Admin Console using the username and password you specified earlier.
+
+. Create a new realm called *myrealm*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-kube#_create_a_realm[create a realm].
+
+. Add a role called *user*. This role will be required to access our simple web application. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#assigning-permissions-using-roles-and-groups[create a role].
+
+. Add a new user named *alice*. Set an *email* address for this new user, we'll use *alice@example.org*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-kube#_create_a_user[create a user].
+
+. Once the new user has been created, set a password for this new user from the *Credentials* tab.
+
+. From the *Role Mapping* tab, assign *alice* the *user* role. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#proc-assigning-role-mappings_server_administration_guide[assign a role] to a user.
+
+. Create a new client as follows:
+* *General Settings*:
+** *Client type* (or *Client Protocol*, depending on your Keycloak version): *OpenID Connect*
+** *Client ID*: *myclient*
+* *Capability config*:
+** *Authentication flow*: *Standard flow*, *Direct access grants*
+* *Login settings*: Leave the fields blank for now.
+
++
+For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#_oidc_clients[Managing OpenID Connect clients].
+
+. Click *Save* to save the client.
+
+== Build a docker image of the application
+
+To build a Docker image from your application so that you can push it to a container repository, such as Docker Hub, follow these steps: 
+
+. Navigate to the application's directory.
+ 
+
+. Use the `wildfly-maven-plugin` plugin to create a docker image.
++
+[source]
+----
+mvn package wildfly:image -Popenshift
+ ...
+[INFO] Successfully tagged localhost/simple-webapp-oidc:latest
+[INFO] acd8c6c41788a99c5b22d2c6b2e42ce024a89e2eb4fe5b4d721f9b56796e95bc
+[INFO] Successfully built application image simple-webapp-oidc:latest
+
+----
+
+. Verify that you see the image in Docker images.
++
+[source]
+----
+docker images
+...
+REPOSITORY                       TAG         IMAGE ID      CREATED        SIZE
+localhost/simple-webapp-oidc     latest      acd8c6c41788  3 minutes ago  690 MB
+----
+
+== Push the just created image to a container registry
+
+To push your image to a container registry so that your application is available to Kubernetes clusters, follow these steps:
+
+. Log in to the container registry.
++
+[source,subs=+quotes]
+----
+docker login __CONTAINER_REGISTRY__ 
```

Comment:
__CONTAINER_REGISTRY__ seems to render a bit odd. Could we use <CONTAINER_REGISTRY> or something similar instead?

Reply:
updated all italics to <form>